### PR TITLE
NCSD-754: CI/CD pipeline + ARM templates

### PR DESCRIPTION
**Pull request by rivoric**
CI/CD pipeline using local ARM templates until the template is considered ready to be moved to dfc-devops

### CODE REVIEW

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            "allowedValues": [
+                "Standard_v2",
+                "WAF_v2"
+            ],
+            "defaultValue": "Standard_v2",
+            "metadata": {
+                "description": "Application gateway tier"
+            }
+        },
+        "backendPools": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: Is this an array of objects?  Should be described in description and\or md file if it is

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+            }
+        },
+        "httpFrontendPort": {
+            "type": "int",
+            "defaultValue": 80,
+            "metadata": {
+                "description": "Http frontend port."
+            }
+        },
+        "httpFrontendPortHttps": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: This is unused, are you planning to use it at a later point?

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+            "allowedValues": [
+                "Enabled",
+                "Disabled"
+            ],
+            "defaultValue": "Disabled",
+            "metadata": {
+                "description": "cookie based affinity"
+            }
+        },
+        "keyVaultName": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: This is unused, are you planning to use it at a later point?

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The name of the key vault."
+            }
+        },
+        "keyVaultSecretName": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: This is unused, are you planning to use it at a later point?

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "The name of the certificate/secret stored in key vault."
+            }
+        },
+        "userAssignedIdentityName": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: This is unused, are you planning to use it at a later point?

File: Resources/ArmTemplates/app-gateway-rewrite.json [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+                "description": "Application gateway tier"
+            }
+        },
+        "backendPools": {
+            "type": "array",
+            "metadata": {
+                "description": "back end pool ip addresses"
+            }
+        },
+        "routingRules": {
```
Comment (NickGraham101) [inline Resources/ArmTemplates/app-gateway-rewrite.json]: Is this an array of objects? Should be described in description and\or md file if it is

File: Resources/ArmTemplates/app-gateway-rewrite.json (hunk 2 above)
Comment (rivoric) [reply Resources/ArmTemplates/app-gateway-rewrite.json]: Got part the way through doing the md file and stopped. Fixed

File: Resources/ArmTemplates/app-gateway-rewrite.json (hunk 3 above)
Comment (rivoric) [reply Resources/ArmTemplates/app-gateway-rewrite.json]: Yes, these were brought across from Craig's template. Will use it to implement the certification. It was just a reminder for me.

File: Resources/ArmTemplates/app-gateway-rewrite.json (hunk 4 above)
Comment (rivoric) [reply Resources/ArmTemplates/app-gateway-rewrite.json]: See above

File: Resources/ArmTemplates/app-gateway-rewrite.json (hunk 6 above)
Comment (rivoric) [reply Resources/ArmTemplates/app-gateway-rewrite.json]: See above

File: Resources/ArmTemplates/app-gateway-rewrite.json (hunk 5 above)
Comment (rivoric) [reply Resources/ArmTemplates/app-gateway-rewrite.json]: Yes, See above

**Pull request merged**